fennel/utils.py

```diff
@@ -76,7 +76,7 @@ def fennel_get_source(obj: Any) -> str:
 def check_response(response: requests.Response):  # type: ignore
     """Check the response from the server and raise an exception if the response is not OK"""
     if response.status_code != 200:
-        print(
+        raise Exception(
```

Comment:
@mohitreddy1996 : why do this? raising an exception causes the program to terminate with a stacktrace that users can't do much about. We recently made this change to go from throwing an exception -> print

Reply:
IMO we should definitely stop/crash the process if a call fails. The main motivation is that, say a sync call fails today, we print something on the terminal and later start executing next calls. e.g. few of the workflows we have seen in the past with Epifi or our load tests is we call sync, sleep for a bit, log data and then attempt to extract features. Say if the sync call fails now, rest of the calls will likely fail. Isn't it a better experience to stop the execution and prompt the user that something failed?

I am coming from CI/CD workflow using Github Actions where as part of a workflow, we sync datasets/features to a dev tier, if it passes, sync in staging and so on.. Sync call fails but I have no way to interrupt the failure. Raising an exception at least allows me to catch the exception and do something of my own (ignore it, handle it, reraise it etc).

Reply:
I agree with what you're saying about interrupting the control flow when a sync error happens and require users to explicitly handle it. My concern has mostly been for the UX of seeing a stacktrace that doesn't make any sense to you. A better idea might be for us to catch these exception in the `sync` function and re-throw it so stack trace looks simpler. wdyt?